Support Boost 1.88

## Describe the bug

Icinga doesn't compile with Boost 1.88.

## To Reproduce

1. Bump Boost version
2. Compile Icinga (like our GHA would do)

```patch
diff --git a/tools/win32/configure.ps1 b/tools/win32/configure.ps1
index 52d8628a1..bd6b42a93 100644
--- a/tools/win32/configure.ps1
+++ b/tools/win32/configure.ps1
@@ -36,10 +36,10 @@ if (-not (Test-Path env:OPENSSL_ROOT_DIR)) {
   $env:OPENSSL_ROOT_DIR = "c:\local\OpenSSL_3_0_15-Win${env:BITS}"
 }
 if (-not (Test-Path env:BOOST_ROOT)) {
-  $env:BOOST_ROOT = "c:\local\boost_1_87_0-Win${env:BITS}"
+  $env:BOOST_ROOT = "c:\local\boost_1_88_0-Win${env:BITS}"
 }
 if (-not (Test-Path env:BOOST_LIBRARYDIR)) {
-  $env:BOOST_LIBRARYDIR = "c:\local\boost_1_87_0-Win${env:BITS}\lib${env:BITS}-msvc-14.2"
+  $env:BOOST_LIBRARYDIR = "c:\local\boost_1_88_0-Win${env:BITS}\lib${env:BITS}-msvc-14.2"
 }
 if (-not (Test-Path env:FLEX_BINARY)) {
   $env:FLEX_BINARY = 'C:\ProgramData\chocolatey\bin\win_flex.exe'
```

## Expected behavior

It compiles.

## Screenshots

```
PS C:\Users\aklimov\icinga2> & powershell.exe .\tools\win32\build.ps1
DEBUG:    1+  >>>> & powershell.exe .\tools\win32\build.ps1
DEBUG:    3+ if ( >>>> -not (Test-Path env:ICINGA2_BUILDPATH)) {
DEBUG:    4+    >>>> $env:ICINGA2_BUILDPATH = '.\build'
DEBUG:    7+ if ( >>>> -not (Test-Path env:CMAKE_BUILD_TYPE)) {
DEBUG:    8+    >>>> $env:CMAKE_BUILD_TYPE = 'RelWithDebInfo'
DEBUG:   11+ if ( >>>> -not (Test-Path $env:ICINGA2_BUILDPATH)) {
DEBUG:   16+ if ( >>>> -not (Test-Path env:CMAKE_PATH)) {
DEBUG:   17+    >>>> $env:CMAKE_PATH = 'C:\Program Files\CMake\bin'
DEBUG:   19+ if ( >>>> -not ($env:PATH -contains $env:CMAKE_PATH)) {
DEBUG:   20+    >>>> $env:PATH = $env:CMAKE_PATH + ';' + $env:PATH
DEBUG:   23+  >>>> cmake.exe --build "$env:ICINGA2_BUILDPATH" --target ALL_BUILD --config $env:CMAKE_BUILD_TYPE
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  1>Checking Build System
  [BISON][class_parser] Building parser with bison 3.7.4
  C:/Users/aklimov/icinga2/tools/mkclass/class_parser.yy:19.1-12: warning: deprecated directive: '%pure-parser', use '%define api.pure' [-Wdeprecated]
     19 | %pure-parser
        | ^~~~~~~~~~~~
        | %define api.pure
  C:/Users/aklimov/icinga2/tools/mkclass/class_parser.yy:23.1-14: warning: deprecated directive: '%error-verbose', use '%define parse.error verbose' [-Wdeprecated]
     23 | %error-verbose
        | ^~~~~~~~~~~~~~
        | %define parse.error verbose
C:/Users/aklimov/icinga2/tools/mkclass/class_parser.yy : warning : 2 shift/reduce conflicts [-Wconflicts-sr] [C:\Users\aklimov\icinga2\Build\tools\mkclass\mkclass.vcxproj]
  C:/Users/aklimov/icinga2/tools/mkclass/class_parser.yy: note: rerun with option '-Wcounterexamples' to generate conflict counterexamples
C:/Users/aklimov/icinga2/tools/mkclass/class_parser.yy : warning : fix-its can be applied.  Rerun with option '--update'. [-Wother] [C:\Users\aklimov\icinga2\Build\tools\mkclass\mkclass.vcxproj]
  [FLEX][class_lexer] Building scanner with win_flex 2.6.4
  Building Custom Rule C:/Users/aklimov/icinga2/tools/mkclass/CMakeLists.txt
  mkclass.cpp
  classcompiler.cpp
  class_lexer.cc
  class_parser.cc
  Code wird generiert...
  mkclass.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\mkclass.exe
  Building Custom Rule C:/Users/aklimov/icinga2/tools/mkunity/CMakeLists.txt
  mkunity.c
  mkunity.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\mkunity.exe
  Generating application-ti.cpp, application-ti.hpp
  Generating configobject-ti.cpp, configobject-ti.hpp
  Generating configuration-ti.cpp, configuration-ti.hpp
  Generating datetime-ti.cpp, datetime-ti.hpp
  Generating filelogger-ti.cpp, filelogger-ti.hpp
  Generating function-ti.cpp, function-ti.hpp
  Generating journaldlogger-ti.cpp, journaldlogger-ti.hpp
  Generating logger-ti.cpp, logger-ti.hpp
  Generating perfdatavalue-ti.cpp, perfdatavalue-ti.hpp
  Generating streamlogger-ti.cpp, streamlogger-ti.hpp
  Generating sysloglogger-ti.cpp, sysloglogger-ti.hpp
  Generating windowseventloglogger-ti.cpp, windowseventloglogger-ti.hpp
  Generating windowseventloglogger-provider.rc, windowseventloglogger-provider.h
  MC: Compiling C:/Users/aklimov/icinga2/lib/base/windowseventloglogger-provider.mc
  Generating base_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/lib/base/CMakeLists.txt
  application-version.cpp
  journaldlogger.cpp
  base_unity.cpp
  Code wird generiert...
  base.vcxproj -> C:\Users\aklimov\icinga2\Build\lib\base\base.dir\RelWithDebInfo\base.lib
  [BISON][config_parser] Building parser with bison 3.7.4
  C:/Users/aklimov/icinga2/lib/config/config_parser.yy:59.1-12: warning: deprecated directive: '%pure-parser', use '%define api.pure' [-Wdeprecated]
     59 | %pure-parser
        | ^~~~~~~~~~~~
        | %define api.pure
  C:/Users/aklimov/icinga2/lib/config/config_parser.yy:63.1-14: warning: deprecated directive: '%error-verbose', use '%define parse.error verbose' [-Wdeprecated]
     63 | %error-verbose
        | ^~~~~~~~~~~~~~
        | %define parse.error verbose
C:/Users/aklimov/icinga2/lib/config/config_parser.yy : warning : 52 shift/reduce conflicts [-Wconflicts-sr] [C:\Users\aklimov\icinga2\Build\lib\config\config.vcxproj]
C:/Users/aklimov/icinga2/lib/config/config_parser.yy : warning : 105 reduce/reduce conflicts [-Wconflicts-rr] [C:\Users\aklimov\icinga2\Build\lib\config\config.vcxproj]
  C:/Users/aklimov/icinga2/lib/config/config_parser.yy: note: rerun with option '-Wcounterexamples' to generate conflict counterexamples
C:/Users/aklimov/icinga2/lib/config/config_parser.yy : warning : fix-its can be applied.  Rerun with option '--update'. [-Wother] [C:\Users\aklimov\icinga2\Build\lib\config\config.vcxproj]
  [FLEX][config_lexer] Building scanner with win_flex 2.6.4
  Generating config_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/lib/config/CMakeLists.txt
  config_lexer.cc
  config_parser.cc
  config_unity.cpp
  Code wird generiert...
C:\Users\aklimov\icinga2\lib\config\expression.cpp(552): warning C4715: "icinga::GetScopeExpression::DoEvaluate": Nicht alle Codepfade geben einen Wert zurück. [C:\Users\aklimov\icinga2\Build\lib\config\config.vcxproj]
  config.vcxproj -> C:\Users\aklimov\icinga2\Build\lib\config\config.dir\RelWithDebInfo\config.lib
  Generating apilistener-ti.cpp, apilistener-ti.hpp
  Generating apiuser-ti.cpp, apiuser-ti.hpp
  Generating endpoint-ti.cpp, endpoint-ti.hpp
  Generating zone-ti.cpp, zone-ti.hpp
  Generating remote_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/lib/remote/CMakeLists.txt
  remote_unity.cpp
  remote.vcxproj -> C:\Users\aklimov\icinga2\Build\lib\remote\remote.dir\RelWithDebInfo\remote.lib
  Building Custom Rule C:/Users/aklimov/icinga2/tools/mkembedconfig/CMakeLists.txt
  mkembedconfig.c
  mkembedconfig.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\mkembedconfig.exe
  Generating icinga-itl.cpp
  Generating checkable-ti.cpp, checkable-ti.hpp
  Generating checkcommand-ti.cpp, checkcommand-ti.hpp
  Generating checkresult-ti.cpp, checkresult-ti.hpp
  Generating command-ti.cpp, command-ti.hpp
  Generating comment-ti.cpp, comment-ti.hpp
  Generating customvarobject-ti.cpp, customvarobject-ti.hpp
  Generating dependency-ti.cpp, dependency-ti.hpp
  Generating downtime-ti.cpp, downtime-ti.hpp
  Generating eventcommand-ti.cpp, eventcommand-ti.hpp
  Generating host-ti.cpp, host-ti.hpp
  Generating hostgroup-ti.cpp, hostgroup-ti.hpp
  Generating icingaapplication-ti.cpp, icingaapplication-ti.hpp
  Generating notification-ti.cpp, notification-ti.hpp
  Generating notificationcommand-ti.cpp, notificationcommand-ti.hpp
  Generating scheduleddowntime-ti.cpp, scheduleddowntime-ti.hpp
  Generating service-ti.cpp, service-ti.hpp
  Generating servicegroup-ti.cpp, servicegroup-ti.hpp
  Generating timeperiod-ti.cpp, timeperiod-ti.hpp
  Generating user-ti.cpp, user-ti.hpp
  Generating usergroup-ti.cpp, usergroup-ti.hpp
  Generating icinga_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/lib/icinga/CMakeLists.txt
  icinga-itl.cpp
  icinga_unity.cpp
  Code wird generiert...
C:\Users\aklimov\icinga2\lib\icinga\checkable-flapping.cpp(114): warning C4715: "icinga::Checkable::ServiceStateToFlappingFilter": Nicht alle Codepfade geben einen Wert zurück. [C:\Users\aklimov\icinga2\Build\lib\icinga\icinga.vcxproj]
C:\Users\aklimov\icinga2\lib\icinga\notification.cpp(640): warning C4715: "icinga::HostStateToFilter": Nicht alle Codepfade geben einen Wert zurück. [C:\Users\aklimov\icinga2\Build\lib\icinga\icinga.vcxproj]
C:\Users\aklimov\icinga2\lib\icinga\notification.cpp(628): warning C4715: "icinga::ServiceStateToFilter": Nicht alle Codepfade geben einen Wert zurück. [C:\Users\aklimov\icinga2\Build\lib\icinga\icinga.vcxproj]
C:\Users\aklimov\icinga2\lib\icinga\notification.cpp(729): warning C4715: "icinga::Notification::NotificationHostStateToString": Nicht alle Codepfade geben einen Wert zurück. [C:\Users\aklimov\icinga2\Build\lib\icinga\icinga.vcxproj]
C:\Users\aklimov\icinga2\lib\icinga\notification.cpp(717): warning C4715: "icinga::Notification::NotificationServiceStateToString": Nicht alle Codepfade geben einen Wert zurück. [C:\Users\aklimov\icinga2\Build\lib\icinga\icinga.vcxproj]
  icinga.vcxproj -> C:\Users\aklimov\icinga2\Build\lib\icinga\icinga.dir\RelWithDebInfo\icinga.lib
  Generating checkercomponent-ti.cpp, checkercomponent-ti.hpp
  Generating checker_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/lib/checker/CMakeLists.txt
  checker_unity.cpp
  checker.vcxproj -> C:\Users\aklimov\icinga2\Build\lib\checker\checker.dir\RelWithDebInfo\checker.lib
  Generating cli_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/lib/cli/CMakeLists.txt
  cli_unity.cpp
  cli.vcxproj -> C:\Users\aklimov\icinga2\Build\lib\cli\cli.dir\RelWithDebInfo\cli.lib
  Generating windowseventloglogger-provider.rc, windowseventloglogger-provider.h
  MC: Compiling C:/Users/aklimov/icinga2/lib/base/windowseventloglogger-provider.mc
  Building Custom Rule C:/Users/aklimov/icinga2/lib/base/CMakeLists.txt
  Generating windowseventloglogger-provider.res
  Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384

  Copyright (C) Microsoft Corporation.  All rights reserved.


  eventprovider.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\eventprovider.dll
  Building Custom Rule C:/Users/aklimov/icinga2/icinga-app/CMakeLists.txt
  icinga.cpp
  icingaloader.vcxproj -> C:\Users\aklimov\icinga2\Build\icinga-app\icingaloader.dir\RelWithDebInfo\icingaloader.lib
  Generating methods-itl.cpp
  Generating methods_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/lib/methods/CMakeLists.txt
  methods-itl.cpp
  methods_unity.cpp
  Code wird generiert...
  methods.vcxproj -> C:\Users\aklimov\icinga2\Build\lib\methods\methods.dir\RelWithDebInfo\methods.lib
  Building Custom Rule C:/Users/aklimov/icinga2/third-party/mmatch/CMakeLists.txt
  mmatch.c
  mmatch.vcxproj -> C:\Users\aklimov\icinga2\Build\third-party\mmatch\mmatch.dir\RelWithDebInfo\mmatch.lib
  Generating notificationcomponent-ti.cpp, notificationcomponent-ti.hpp
  Generating notification_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/lib/notification/CMakeLists.txt
  notification_unity.cpp
  notification.vcxproj -> C:\Users\aklimov\icinga2\Build\lib\notification\notification.dir\RelWithDebInfo\notification.lib
  Building Custom Rule C:/Users/aklimov/icinga2/third-party/socketpair/CMakeLists.txt
  socketpair.c
  socketpair.vcxproj -> C:\Users\aklimov\icinga2\Build\third-party\socketpair\socketpair.dir\RelWithDebInfo\socketpair.lib
  Generating base_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/test/CMakeLists.txt
  test-runner.cpp
  base_unity.cpp
C:\Users\aklimov\icinga2\test/base-stacktrace.cpp(23,9): warning C4068: Unbekanntes Pragma "GCC". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-base.vcxproj]
C:\Users\aklimov\icinga2\test/base-stacktrace.cpp(24,9): warning C4068: Unbekanntes Pragma "GCC". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-base.vcxproj]
C:\Users\aklimov\icinga2\test/base-stacktrace.cpp(25,9): warning C4068: Unbekanntes Pragma "clang". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-base.vcxproj]
C:\Users\aklimov\icinga2\test/base-stacktrace.cpp(68,9): warning C4068: Unbekanntes Pragma "GCC". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-base.vcxproj]
C:\Users\aklimov\icinga2\test/base-stacktrace.cpp(69,9): warning C4068: Unbekanntes Pragma "clang". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-base.vcxproj]
  Code wird generiert...
     Bibliothek "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/boosttest-test-base.lib" und Objekt "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/boosttest-test-base.exp" werden erstellt.
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "EnumProcessModules" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail@s
tacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-base.vcxproj]
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "GetModuleInformation" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail
@stacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-base.vcxproj]
C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\boosttest-test-base.exe : fatal error LNK1120: 2 nicht aufgelöste Externe [C:\Users\aklimov\icinga2\Build\test\boosttest-test-base.vcxproj]
  Generating icinga_checkable_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/test/CMakeLists.txt
  test-runner.cpp
  icinga_checkable_unity.cpp
  Code wird generiert...
     Bibliothek "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/boosttest-test-icinga_checkable.lib" und Objekt "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/boosttest-test-icinga_checkable.exp" wer
  den erstellt.
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "EnumProcessModules" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail@s
tacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-icinga_checkable.vcxproj]
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "GetModuleInformation" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail
@stacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-icinga_checkable.vcxproj]
C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\boosttest-test-icinga_checkable.exe : fatal error LNK1120: 2 nicht aufgelöste Externe [C:\Users\aklimov\icinga2\Build\test\boosttest-test-icinga_checkable.vcxproj]
  Generating types_unity.cpp
  Building Custom Rule C:/Users/aklimov/icinga2/test/CMakeLists.txt
  test-runner.cpp
  types_unity.cpp
  Code wird generiert...
     Bibliothek "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/boosttest-test-types.lib" und Objekt "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/boosttest-test-types.exp" werden erstellt.
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "EnumProcessModules" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail@s
tacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-types.vcxproj]
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "GetModuleInformation" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail
@stacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\test\boosttest-test-types.vcxproj]
C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\boosttest-test-types.exe : fatal error LNK1120: 2 nicht aufgelöste Externe [C:\Users\aklimov\icinga2\Build\test\boosttest-test-types.vcxproj]
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  thresholds.cpp
  thresholds.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\thresholds.lib
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_disk.cpp
  check_disk.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_disk.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_load.cpp
  check_load.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_load.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_memory.cpp
  check_memory.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_memory.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_network.cpp
  check_network.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_network.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_nscp_api.cpp
     Bibliothek "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/check_nscp_api.lib" und Objekt "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/check_nscp_api.exp" werden erstellt.
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "EnumProcessModules" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail@s
tacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\plugins\check_nscp_api.vcxproj]
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "GetModuleInformation" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail
@stacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\plugins\check_nscp_api.vcxproj]
C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_nscp_api.exe : fatal error LNK1120: 2 nicht aufgelöste Externe [C:\Users\aklimov\icinga2\Build\plugins\check_nscp_api.vcxproj]
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_perfmon.cpp
  check_perfmon.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_perfmon.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_ping.cpp
  check_ping.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_ping.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_procs.cpp
  check_procs.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_procs.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_service.cpp
  check_service.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_service.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_swap.cpp
  check_swap.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_swap.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_update.cpp
  check_update.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_update.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_uptime.cpp
  check_uptime.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_uptime.exe
  Building Custom Rule C:/Users/aklimov/icinga2/plugins/CMakeLists.txt
  check_users.cpp
  check_users.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\check_users.exe
  Building Custom Rule C:/Users/aklimov/icinga2/icinga-app/CMakeLists.txt
     Bibliothek "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/icinga2.lib" und Objekt "C:/Users/aklimov/icinga2/Build/Bin/RelWithDebInfo/RelWithDebInfo/icinga2.exp" werden erstellt.
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "EnumProcessModules" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail@s
tacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\icinga-app\icinga-app.vcxproj]
base_unity.obj : error LNK2019: Verweis auf nicht aufgelöstes externes Symbol "GetModuleInformation" in Funktion ""unsigned __int64 __cdecl boost::stacktrace::detail::get_own_proc_addr_base(void const *)" (?get_own_proc_addr_base@detail
@stacktrace@boost@@YA_KPEBX@Z)". [C:\Users\aklimov\icinga2\Build\icinga-app\icinga-app.vcxproj]
C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\icinga2.exe : fatal error LNK1120: 2 nicht aufgelöste Externe [C:\Users\aklimov\icinga2\Build\icinga-app\icinga-app.vcxproj]
  Building Custom Rule C:/Users/aklimov/icinga2/icinga-installer/CMakeLists.txt
  icinga-installer.cpp
  icinga-installer.vcxproj -> C:\Users\aklimov\icinga2\Build\Bin\RelWithDebInfo\RelWithDebInfo\icinga2-installer.exe
  Icinga2SetupAgent -> C:\Users\aklimov\icinga2\agent\windows-setup-agent\bin\RelWithDebInfo\Icinga2SetupAgent.exe
DEBUG:   24+ if ( >>>> $lastexitcode -ne 0) { exit $lastexitcode }
DEBUG:   24+ if ($lastexitcode -ne 0) {  >>>> exit $lastexitcode }
PS C:\Users\aklimov\icinga2>
```

## Your Environment

* Version used (`icinga2 --version`): 520aed6049c6a6142fbe322bfc9b20214bae7290
* Operating System: Windows

--- a/doc/win-dev.ps1
+++ b/doc/win-dev.ps1
@@ -13,7 +13,7 @@ function ThrowOnNativeFailure {
 
 $VsVersion = 2019
 $MsvcVersion = '14.2'
-$BoostVersion = @(1, 87, 0)
+$BoostVersion = @(1, 88, 0)
 $OpensslVersion = '3_0_16'
 
 switch ($Env:BITS) {

--- a/lib/base/win32.hpp
+++ b/lib/base/win32.hpp
@@ -5,7 +5,7 @@
 
 #define WIN32_LEAN_AND_MEAN
 #ifndef _WIN32_WINNT
-#define _WIN32_WINNT _WIN32_WINNT_VISTA
+#define _WIN32_WINNT _WIN32_WINNT_WIN7
 #endif /* _WIN32_WINNT */
 #define NOMINMAX
 #include <winsock2.h>

--- a/tools/win32/configure-dev.ps1
+++ b/tools/win32/configure-dev.ps1
@@ -34,10 +34,10 @@ if (-not (Test-Path env:OPENSSL_ROOT_DIR)) {
   $env:OPENSSL_ROOT_DIR = 'c:\local\OpenSSL-Win64'
 }
 if (-not (Test-Path env:BOOST_ROOT)) {
-  $env:BOOST_ROOT = 'c:\local\boost_1_87_0'
+  $env:BOOST_ROOT = 'c:\local\boost_1_88_0'
 }
 if (-not (Test-Path env:BOOST_LIBRARYDIR)) {
-  $env:BOOST_LIBRARYDIR = 'c:\local\boost_1_87_0\lib64-msvc-14.2'
+  $env:BOOST_LIBRARYDIR = 'c:\local\boost_1_88_0\lib64-msvc-14.2'
 }
 if (-not (Test-Path env:FLEX_BINARY)) {
   $env:FLEX_BINARY = 'C:\ProgramData\chocolatey\bin\win_flex.exe'

--- a/tools/win32/configure.ps1
+++ b/tools/win32/configure.ps1
@@ -36,10 +36,10 @@ if (-not (Test-Path env:OPENSSL_ROOT_DIR)) {
   $env:OPENSSL_ROOT_DIR = "c:\local\OpenSSL_3_0_16-Win${env:BITS}"
 }
 if (-not (Test-Path env:BOOST_ROOT)) {
-  $env:BOOST_ROOT = "c:\local\boost_1_87_0-Win${env:BITS}"
+  $env:BOOST_ROOT = "c:\local\boost_1_88_0-Win${env:BITS}"
 }
 if (-not (Test-Path env:BOOST_LIBRARYDIR)) {
-  $env:BOOST_LIBRARYDIR = "c:\local\boost_1_87_0-Win${env:BITS}\lib${env:BITS}-msvc-14.2"
+  $env:BOOST_LIBRARYDIR = "c:\local\boost_1_88_0-Win${env:BITS}\lib${env:BITS}-msvc-14.2"
 }
 if (-not (Test-Path env:FLEX_BINARY)) {
   $env:FLEX_BINARY = 'C:\ProgramData\chocolatey\bin\win_flex.exe'

